packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -165,6 +192,34 @@ export class PendingStateManager {
         this.pendingStates.push(pendingFlush);
     }
 
+    public processRemoteMessage(message: ISequencedDocumentMessage) {
+        while (!this.initialStates.isEmpty()) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const nextMessage = this.initialStates.peekFront()!;
+            assert(nextMessage.type === "message"); // assume it's not flush for now
+            if (nextMessage.referenceSequenceNumber > message.sequenceNumber) {
+                break;
+            }
+            const context = (this.containerRuntime as any).dataStores.contexts.get(nextMessage.content?.address);
+            const channelContext = context.channel.contexts.get(nextMessage.content.contents?.content?.address);
+
+            // rebaseOp will cause the DDS to behave as if it has sent the op but not actually send it
+            channelContext.rebaseOp(nextMessage.content.contents.content, nextMessage.localOpMetadata);
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            this.pendingRebase.push(this.initialStates.shift()!);
+        }
+        if (message.clientId === this.initialClientId && message.clientSequenceNumber >= this.initialClientSeqNum) {
+            // must be a pending op that was successfully sent
+            assert(!this.pendingRebase.isEmpty());
```

Comment:
better remove it and replace with assert(ackedMessage  !== undefined) below - you would not need to use ! and silence compiler